### DISCUSSION TITLE
can‘t found streams on twitch

### DISCUSSION THREAD

**Discussion by agafadsf** [streamlink/streamlink#4150] [Q&A] [answered]
when I use steamlink to get streams on twitch , it always  says
error: No playable streams found on this URL

**Comment by bastimeyer** [comment, marked as answer]
Which kind of replies do you expect from posting this? You didn't provide any information at all, so nobody will be able to help.

**Post the entire command you're trying to run with the entire output, and include the `--loglevel=debug` parameter.**

**Reply by agafadsf** [reply]
sry,
input： streamlink.exe https://www.twitch.tv/gaules
output： [cli][info] Found matching plugin twitch for URL https://www.twitch.tv/gaules
error: No playable streams found on this URL: https://www.twitch.tv/gaules

**Reply by beardypig** [reply]
We need to see the full output when using `--loglevel=debug`, it includes a lot of information that will help debug your issue - notable the version of streamlink you are using.

**Reply by beardypig** [reply]
I'm going to assume you are using an old version of Streamlink, you should update to the latest version and report back if you still have the issue.

**Reply by agafadsf** [reply]
[cli][debug] OS:         Windows 10
[cli][debug] Python:     3.9.6
[cli][debug] Streamlink: 2.4.0
[cli][debug] Requests(2.26.0), Socks(1.7.1), Websocket(1.2.1)
[cli][debug] Arguments:
[cli][debug]  url=https://www.twitch.tv/gaules
[cli][debug]  --loglevel=debug
[cli][debug]  --player="C:\Program Files (x86)\DAUM\PotPlayer\PotPlayerMini.exe"
[cli][debug]  --rtmp-rtmpdump=rtmpdump.exe
[cli][debug]  --rtmpdump=F:\streamlink\rtmpdump\rtmpdump.exe
[cli][debug]  --ffmpeg-ffmpeg=F:\streamlink\ffmpeg\ffmpeg.exe
[cli][info] Found matching plugin twitch for URL https://www.twitch.tv/gaules
[plugins.twitch][debug] Getting live HLS streams for gaules
error: No playable streams found on this URL: https://www.twitch.tv/gaules

**Reply by beardypig** [reply]
@agafadsf try the latest nightly version, which installer did you use?

**Reply by agafadsf** [reply]
https://github.com/streamlink/streamlink/releases/tag/2.4.0
i use the installer from this link

**Reply by bastimeyer** [reply]
> [plugins.twitch][debug] Getting live HLS streams for gaules

This implies that everything until the stream access token request was successful, and then it fails, probably due to the `None` return value here:
https://github.com/streamlink/streamlink/blob/2.4.0/src/streamlink/plugins/twitch.py#L597-L598

Has Twitch blocked your IP address? A 404 HTTP status shouldn't occur when requesting the HLS playlist and if it does, then Twitch is doing this most likely intentionally. An HLS playlist parsing error basically means the same, because Twitch is returning different data, probably because they've blocked you.

**What happens if you open the stream in your web browser?**

**Reply by agafadsf** [reply]
That makes sense. I'm from China, so maybe my IP address is blocked. It seems that vpn doesn't work on streamlink. Is there anything i can do to solve this problem? 
Thx for your help

**Reply by bastimeyer** [reply]
You could try using HTTP or SOCKS proxies if your system's VPN configuration is not working.
https://streamlink.github.io/latest/cli.html#proxy-support
However, this is not a Streamlink issue and you'll have to figure it out on your own. Good luck.
